styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -570,11 +571,28 @@ private void setupMetrics(
           workflow.id(),
           () -> stateManager.getActiveStatesCount(workflow.id()));
 
+      final Optional<Workflow> existingWorkflow = cache.workflow(workflow.id());
+      if (existingWorkflow.isPresent()) {
+        if (!isGreaterOrEqualApiVersion(workflow, existingWorkflow.get())) {
+          return;
+        }
+      }
+
       cache.store(workflow);
       workflowInitializer.inspectChange(workflow);
     };
   }
 
+  // if we have ever registered/updated the workflow via API,
+  // we ignore any subsequent updates via schedule source or lower API version
```

Comment:
This describes "what" but doesn't describe "why". Harrumph. "We ignore _in order to_ enable gradual rollout of workflow configuration API endpoint, while still supporting the polling schedule source for users that haven't migrated".